src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -343,10 +343,17 @@ private class RsFnInferenceContext(
     }
 
     fun inferLitExprType(expr: RsLitExpr, expected: Ty?): Ty {
```

Comment:
Why do we need these two similar functions? :)

Reply:
Oh, it's another story =D

First works in function inference context, second - in out-of-function context (constanst/statics)

Reply:
Could we unify them?

Reply:
Really we currently very badly support out-of-fn expressions, and it requires detailed consideration. I'll open an issue

Reply:
OK :) In general, I believe such low-hanging refactoring fruit should be picked up as soon as possible, otherwise it tends to turn into wine...